src/utils/weather_model.jl

```diff
@@ -0,0 +1,215 @@
+        to_z_levels(raw_data_path, ic_data_path, target_levels, Float32)
+    end
+    return ic_data_path
+end
+
+
+"""
+    to_z_levels(source_file, target_file, target_levels, FT)
+
+Interpolate ERA5 data from native model levels to specified z-levels. Note that 
+to match MoistFromFile, we rename surface pressure, sp, to p. This should change 
+for consistency if we write a new initial condition dispatch.
+
+# Arguments
+- `source_file` / `target_file`: Path to input / output NetCDF 
+- `target_levels`: Vector of target altitude levels (in meters)
+
+"""
+function to_z_levels(source_file, target_file, target_levels, FT)
+
+    param_set = TD.Parameters.ThermodynamicsParameters(FT)
+    grav = TD.Parameters.grav(param_set)
+
+    ncin = Dataset(source_file)
+
+    # Read and cast coordinates to FT type
+    lat = FT.(ncin["latitude"][:])
+    lon = FT.(ncin["longitude"][:])
+
+    # Read and cast variables to FT type, replacing missing values with NaN
+    z_raw = ncin["z"][:, :, :, 1]
+    source_z = FT.(coalesce.(z_raw, NaN)) ./ grav # convert from geopotential height to height
+
+    # Create output file
+    ncout = NCDataset(target_file, "c", attrib = copy(ncin.attrib))
+
+    # Define dimensions
+    defDim(ncout, "lon", size(lon, 1))
+    defDim(ncout, "lat", size(lat, 1))
+    defDim(ncout, "z", length(target_levels))
+
+    # Define coordinate variables with clean attributes
+    lon_attrib = Dict(
+        "Datatype" => string(FT),
+        "standard_name" => "longitude",
+        "long_name" => "longitude",
+        "units" => "degrees_east",
+    )
+    lon_var = defVar(ncout, "lon", FT, ("lon",), attrib = lon_attrib)
+    lon_var[:] = lon
+
+    lat_attrib = Dict(
+        "Datatype" => string(FT),
+        "standard_name" => "latitude",
+        "long_name" => "latitude",
+        "units" => "degrees_north",
+        "stored_direction" => "decreasing",
+    )
+    lat_var = defVar(ncout, "lat", FT, ("lat",), attrib = lat_attrib)
+    lat_var[:] = lat
+
+    z_attrib = Dict(
+        "Datatype" => string(FT),
+        "standard_name" => "altitude",
+        "long_name" => "altitude",
+        "units" => "m",
+    )
+    z_var = defVar(ncout, "z", FT, ("z",), attrib = z_attrib)
+    z_var[:] = target_levels
+
+    # Interpolate and write 3D variables
+    u_var =
+        defVar(ncout, "u", FT, ("lon", "lat", "z"), attrib = ncin["u"].attrib)
+    u_var[:, :, :] =
+        interpz_3d(target_levels, source_z, FT.(ncin["u"][:, :, :, 1]))
+
+    v_var =
+        defVar(ncout, "v", FT, ("lon", "lat", "z"), attrib = ncin["v"].attrib)
+    v_var[:, :, :] =
+        interpz_3d(target_levels, source_z, FT.(ncin["v"][:, :, :, 1]))
+
+    # ERA5 w is from a hydrostatic model and so isn't meaningful for ClimaAtmos
+    # See https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2017MS001059
+    w_var =
+        defVar(ncout, "w", FT, ("lon", "lat", "z"), attrib = ncin["w"].attrib)
+    w_var[:, :, :] .= FT(0)
```

Comment:
You can use `dimnames` instead and I think you can use this everywhere if you are not permuting the dimensions of the array.

```suggestion
        defVar(ncout, "w", FT, dimnames(ncin["w"]), attrib = ncin["w"].attrib)
    w_var[:, :, :] .= FT(0)
```